pimd/pim_msg.h

```diff
@@ -158,10 +161,16 @@ struct pim_encoded_source_ipv6 {
 typedef struct pim_encoded_ipv4_unicast pim_encoded_unicast;
 typedef struct pim_encoded_group_ipv4   pim_encoded_group;
 typedef struct pim_encoded_source_ipv4  pim_encoded_source;
+typedef struct ip                       ipv_hdr;
```

Comment:
Can be done better: create two (for each IP version, in ifdef) `pim_sgaddr pim_sgaddr_from_iphdr(const void *iphdr)` helper functions ⇒ no need for `ipv_hdr` + `IPV_SRC` + `IPV_DST`

Reply:
Fixed.